timemachine/potentials/bonded_stable.py

```diff
@@ -0,0 +1,56 @@
+import jax.numpy as jnp
+
+
+def harmonic_angle_stable(conf, params, angle_idxs, cos_angles=True):
+    r"""
+    Compute the harmonic angle energy using a numerically stable approximation.
+
+    The functional form is identical to :py:func:`potentials.bonded.HarmonicAngle`, except that the following
+    approximation is used for the intermediate computation of :math:`\cos(\theta)`:
+
+    :math::
+
+        \cos(\theta) \approx \frac{r_{ij} \cdot r_{kj} + \epsilon^2}{\sqrt{(r_{ij}^2 + \epsilon^2) (r_{kj}^2 + \epsilon^2)}}
+
+    This reduces to the exact expression when :math:`\epsilon = 0`; When :math:`\epsilon > 0`, this avoids the
+    singularities in the exact expression as :math:`r_{ij}` or :math:`r_{kj}` approach zero.
+
+    Parameters:
+    -----------
+    conf: shape [num_atoms, 3] np.ndarray
+        atomic coordinates
+
+    params: shape [num_angles, 3] np.ndarray
+        force constants, eq angles, epsilons
+        (kas, a0s, epsilons = params.T)
+
+    angle_idxs: shape [num_angles, 3] np.ndarray
+        each element (a, b, c) is a unique angle in the conformation. atom b is defined
+        to be the middle atom.
+
+    cos_angles: True (default)
+        if True, then this instead implements V(t) = k*(cos(t)-cos(t0))^2. This is far more
```

Comment:
nit: k/2 ?

Reply:
Done in https://github.com/proteneer/timemachine/pull/935/commits/eda6999e88d7eb85376c1c5b4cacbdd2ad14d78b